frontend/packages/ceph-storage-plugin/src/components/independent-mode/utils.ts

```diff
@@ -1,27 +1,56 @@
 import * as _ from 'lodash';
+import { Base64 } from 'js-base64';
 import { ClusterServiceVersionKind } from '@console/operator-lifecycle-manager';
 import { HealthState } from '@console/shared/src/components/dashboard/status-card/states';
-import { DataValidator, DataState, ErrorType, Field } from './types';
 import { K8sResourceKind } from '@console/internal/module/k8s';
 
-export const getValidJSON: DataValidator = (fData) => {
+const pluralize = (count: number, singular: string, plural: string = `${singular}s`): string =>
+  count > 1 ? plural : singular;
+
+export const isValidJSON = (fData: string): boolean => {
   try {
-    // Just see if it fails
-    const parsedData = JSON.parse(fData);
-    return { isValid: true, parsedData };
+    JSON.parse(fData);
+    return true;
   } catch (e) {
-    return { isValid: false, errorMessage: 'File is not a valid JSON file.' };
+    return false;
   }
 };
 
-export const checkError = (data: DataState): ErrorType[] => {
-  const errors: ErrorType[] = [];
-  for (const key in data) {
-    if (!_.isNull(data[key]) && _.isEmpty(_.trim(data[key])))
-      errors.push({ field: key as Field, message: `${key} cannot be empty` });
-    else errors.push({ field: key as Field, message: '' });
+export const checkError = (data: string = '{}', requiredKeys = []): string => {
+  const providedKeys = Object.keys(JSON.parse(data || '{}'));
+  const emptyKeys = [];
+  const base64ErrorKeys = [];
+  _.map(JSON.parse(data || '{}'), (v, k) => {
```

Comment:
Better to use javaScript `map` method here

Reply:
Objects do not support maps in Javascript.

Reply:
I thought we are iterating over array. Thanks!